default-recommendations/for-loop-shortcuts.rkt

```diff
@@ -275,6 +275,15 @@
   ((~replacement loop-id.set-id #:original loop-id) clauses body ...))
 
 
+(define-refactoring-rule for/vector-with-in-range-to-length
+  #:description "Add `#:length` to `for/vector` loops to improve performance when the number of iterations is known."
+  #:literals (for/vector in-range)
+  (for/vector
+   (~and for-clauses ([var:id (~and range-expr (in-range (~optional 0) end:id))]))
+   body:expr ...+)
+  (for/vector #:length end ([var range-expr]) body ...))
```

Comment:
You don't need the `range-expr` binding. You only need the `for-clauses` binding.


```suggestion
   (~and for-clauses ([var:id (in-range (~optional 0) end:id)]))
   body:expr ...+)
  (for/vector #:length end for-clauses body ...))
```